mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -159,19 +157,27 @@ impl S3ClientConfig {
 }
 
 /// Authentication configuration for the CRT-based S3 client
-#[derive(Debug, Clone, Default)]
+#[derive(Clone, Debug)]
 pub enum S3ClientAuthConfig {
     /// The default AWS credentials resolution chain, similar to the AWS CLI
-    #[default]
-    Default,
+    DefaultChain {
+        /// Optional profile override to be used when evaluating credential chain
+        profile_name_override: Option<String>,
```

Comment:
naming nit: I think `Option` already captures that it's an override of sorts, so maybe just `profile_name`